src/Common/HostResolvePool.cpp

```diff
@@ -158,8 +160,14 @@ void HostResolver::setFail(const Poco::Net::IPAddress & address)
         if (it == records.end())
             return;
 
-        it->failed = true;
-        it->fail_time = now;
+        while (it != records.end() && it->address == address)
+        {
+            it->failed = true;
+            it->fail_time = now;
+            if (it->fail_count < RECORD_FAIL_COUNT_LIMIT)
+                ++it->fail_count;
+            ++it;
+        }
```

Comment:
what's the point of creating `while` cycle here? We assume that we may have the same addresses there?

Reply:
Yes, you a right, with single record for address this cycle useless, removed.